src/js/ripple/amount.js

```diff
@@ -798,29 +576,41 @@ Amount.prototype.parse_quality = function(quality, counterCurrency, counterIssue
 
   var baseCurrency = Currency.from_json(opts.base_currency);
 
-  this._is_negative = false;
-  this._value       = new BigInteger(quality.substring(quality.length-14), 16);
-  this._offset      = parseInt(quality.substring(quality.length-16, quality.length-14), 16)-100;
+  var mantissa_hex  = quality.substring(quality.length-14);
+  var offset_hex    = quality.substring(quality.length-16, quality.length-14);
+  var mantissa      = new BigNumber(mantissa_hex, 16);
+  var offset        = parseInt(offset_hex, 16) - 100;
+
+  var value = new BigNumber(mantissa.toString() + 'e' + offset.toString());
+
   this._currency    = Currency.from_json(counterCurrency);
   this._issuer      = UInt160.from_json(counterIssuer);
   this._is_native   = this._currency.is_native();
 
+  var exponent = 0;
+  if (this._is_native) {
+    if (opts.inverse) {
+      exponent += 1;
+    } else {
+      exponent -= 1;
```

Comment:
> I'm not opposed to changing the behaviour of this code.

I think this blog post does a decent job of explaining the principle that I'm trying to obey here:
http://jakegoulding.com/blog/2012/11/04/refactor-and-make-changes-in-different-commits/

With regards to overflowing to zero, I misinterpreted your original comment; it should be fixed now.